lib/broadway_rabbitmq/producer.ex

```diff
@@ -357,13 +353,20 @@ defmodule BroadwayRabbitMQ.Producer do
     %{channel: channel, client: client, config: config} = state
     %{delivery_tag: tag, redelivered: redelivered} = meta
 
-    ack_data = %{
-      delivery_tag: tag,
-      client: client,
-      redelivered: redelivered,
-      on_success: state.on_success,
-      on_failure: state.on_failure
-    }
+    acknowledger =
+      if config[:consume_options][:no_ack] do
```

Comment:
Can `config[:consume_options]` ever be `nil` or NimbleOptions will take care of that for us?

Reply:
NimbleOptions sets the default to `[]` so no this is never `nil`